draft-kazuho-httpbis-streaming-requests.md

```diff
@@ -64,26 +82,27 @@ HTTP request downstream before receiving the complete request.
 The term Boolean is imported from {{!STRUCTURED-FIELDS=RFC8941}}.
 
 
-# The Request-Streaming Header Field
+# The Incremental Header Field
 
-The Request-Streaming request header field expresses the client's intent for
-HTTP intermediaries to start forwarding the request to downstream servers before
-the entire request is received.
+The Incremental HTTP header field expresses the sender's intent for HTTP
+intermediaries to start forwarding the message downstream before the entire
+message is received.
 
 This header field has just one valid value of type Boolean: "?1".
 
 ~~~
-Request-Streaming = ?1
+Incremental = ?1
 ~~~
 
-Upon receiving a header section that includes the Request-Streaming header
-field, HTTP intermediaries SHOULD NOT buffer the entire request before
-forwarding it. Instead, intermediaries SHOULD establish an HTTP channel to
-downstream servers, transmit the header section, and continuously forward the
-bytes of the request body as they arrive.
+Upon receiving a header section that includes the Incremental header field, HTTP
+intermediaries SHOULD NOT buffer the entire message before forwarding it.
+Instead, intermediaries SHOULD transmit the header section downstream and
+continuously forward the bytes of the message body as they arrive.
 
-Similarly, intermediaries SHOULD forward the bytes of the response body to the
-clients as they are received.
+The Incremental HTTP header field applies to each HTTP message. Therefore, if
+both the HTTP request and response need to be forwarded incrementally, the
+Incremental HTTP header field MUST be set for both the HTTP request and the
+response.
 
```

Comment:
```suggestion

The Incremental field is advisory. Intermediaries that are unaware of the field
or that do not support the field might buffer messages, even when explicitly
requested otherwise.  Clients and servers therefore cannot expect all
intermediaries to understand and respect a request to deliver messages
incrementally. Clients can rely on prior knowledge or probe for support on
individual resources.
```

(Happy to take that to a separate change; it's not exactly simple.)

Reply:
Let's get this in. As you say we might want to discuss, I think it is a fact that the signal is only advisory; it makes sense to point that out.